packages/caches/src/tests/weak.ts

```diff
@@ -2,16 +2,22 @@ import * as assert from "assert";
 import { WeakCache } from "../weak.js";
 
 describe("weak least-recently-used cache", function () {
-  it("can hold lots of elements", function () {
+  it("can hold lots of elements", async function () {
+    this.timeout(5000);
     const cache = new WeakCache();
     const count = 1000000;
     const keys = [];
 
+    console.time("filling up cache");
     for (let i = 0; i < count; ++i) {
       const key = {};
       cache.set(key, String(i));
       keys[i] = key;
     }
+    console.timeEnd("filling up cache");
+    console.time("waiting for finalization");
+    await waitForCache(cache);
```

Comment:
This makes sense for display reasons, but in the final test we would (probably?) omit this part and just let it happen by itself?